src/tools/opt-dist/src/environment/windows.rs

```diff
@@ -0,0 +1,64 @@
+use crate::environment::Environment;
+use crate::utils::io::move_directory;
+use camino::Utf8PathBuf;
+use std::io::{Cursor, Read};
+use zip::ZipArchive;
+
+pub(super) struct WindowsEnvironment {
+    checkout_dir: Utf8PathBuf,
+}
+
+impl WindowsEnvironment {
+    pub fn new() -> Self {
+        Self { checkout_dir: std::env::current_dir().unwrap().try_into().unwrap() }
+    }
+}
+
+impl Environment for WindowsEnvironment {
+    fn checkout_path(&self) -> Utf8PathBuf {
+        self.checkout_dir.clone()
+    }
+
+    fn downloaded_llvm_dir(&self) -> Utf8PathBuf {
+        self.checkout_path().join("citools/clang-rust")
+    }
+
+    fn opt_artifacts(&self) -> Utf8PathBuf {
+        self.checkout_path().join("opt-artifacts")
+    }
+
+    fn build_root(&self) -> Utf8PathBuf {
+        self.checkout_path()
+    }
+
+    fn prepare_rustc_perf(&self) -> anyhow::Result<()> {
+        const PERF_COMMIT: &str = "9dfaa35193154b690922347ee1141a06ec87a199";
```

Comment:
Easy to desync commit with linux version, for example like now.

Reply:
Good catch, but this is a pre-existing issue. Desync is not such a big problem here, and might even be needed (e.g. if something stops compiling only on one platform). It would be nice if we could somehow resolve this, but since the other value is in a Dockerfile, it might be a bit tricky, I probably don't want to stuff the commit into the GHA workflow YAML file or something like that.

Reply:
At least FIXME with description that this should be somehow synced.

Reply:
Good point, I will add it. I will not change the commit of the Windows version in this PR though, to avoid mixing that with the rewrite.

Reply:
Ooh, the version *has* changed in the meantime (I started this in March?!), I didn't realize that it is actually desynced in the PR. Thank you!

Reply:
I wrote comment that can be wrongly interpreted, my bad.